src/Workspaces/CSharpTest/Formatting/FormattingTests.cs

```diff
@@ -9498,5 +9498,35 @@ enum TestEnum
 }", changedOptionSet: changingOptions);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        [WorkItem(39433, "https://github.com/dotnet/roslyn/issues/39433")]
+        public async Task LambdaInsideTupleStaysOnSameLine()
```

Comment:
I don't know if this is the right location for a test like this, since it actually doesn't changes the formatting in this scenario, but I couldn't find a better location.

Next, the `input`, and `expected` source are identical, since I couldn't find the way how to assert that. Any suggestions?

Reply:
❔ Does this test fail prior to your change? Or more specifically, does this test fail if the change to **TokenBasedFormattingRule.cs** isn't present?

Reply:
Yeah, I first tried to reproduce the issue using [this commit](https://github.com/tiesmaster/roslyn/commit/9d6dacf68ceba0a99f393e74e08418caf075fa7d) (which fails), and then fixed the issue. I rebased that commit away, because I didn't thought it'd be that helpful, but I can keep that kind of work, if it helps the PR.